libcontainer/container_linux.go

```diff
@@ -276,6 +276,21 @@ func (c *linuxContainer) Exec() error {
 func (c *linuxContainer) exec() error {
 	path := filepath.Join(c.root, execFifoFilename)
 
+	defer func() {
+		if c.config.Hooks != nil && len(c.config.Hooks.Poststart) > 0 {
+			s, err := c.currentOCIState()
+			if err != nil {
+				return
+			}
+			for i, hook := range c.config.Hooks.Poststart {
+				if err := hook.Run(s); err != nil {
+					fmt.Printf("%v", newSystemErrorWithCausef(err, "running poststart hook %d", i))
```

Comment:
Printing (to stdout?) is not a great API for callers.  Can we collect these and return any failed-hook [warnings][1] to the caller?

[1]: https://github.com/opencontainers/runtime-spec/blame/v1.0.1/runtime.md#L63

Reply:
@wking Right. I'll make it return warnings to the caller.